### DISCUSSION TITLE
Are web sockets exposed to the stack of middleware?

### DISCUSSION THREAD

**Discussion by yatesco** [tokio-rs/axum#614] [Q&A] [answered]
I assume not. I assume a web socket is an opaque tunnel as far as Axum's middleware is concerned (other than the initial request to upgrade). Is that correct?

My use case is enforcing authentication on each "request", whereas a web socket message isn't a "request".

**Comment by davidpdrsn** [comment, marked as answer]
Correct.

The client makes a regular HTTP request that is upgraded to a websocket connection at which point middleware are no longer involved.

**Reply by yatesco** [reply]
thanks David (you are on fire this morning!)
